src/dkg/key_generation.rs

```diff
@@ -883,15 +889,14 @@ impl<C: CipherSuite> DistributedKeyGeneration<RoundOne, C> {
                     let decrypted_share = decrypt_share(encrypted_share, &dh_key_bytes);
                     let decrypted_share_ref = &decrypted_share;
 
-                    for commitment in self.state.their_commitments.as_ref().unwrap().iter() {
+                    for commitment in self.state.their_commitments.as_ref().expect("Dealers always have commitments to their secret polynomial evaluations.").iter() {
                         if commitment.index == encrypted_share.sender_index {
-                            // If the decrypted share is incorrect, P_i builds
-                            // a complaint
+                            // If the decrypted share is incorrect, P_i builds a complaint.
 
                             if decrypted_share.is_err()
                                 || decrypted_share_ref
                                     .as_ref()
-                                    .unwrap()
+                                    .expect("This cannot fail.")
```

Comment:
Well, this literally cannot fail. We're evaluating the second operand lazily, so if we're reaching the unwrap then we will have something inside

Reply:
The lazy evaluation is not on the same element though, is it? What if decrypted_share_ref somehow doesn't correspond to a reference to decrypted_share? (I mean, I know it shouldn't happen, but if there is a discrepancy, there is a problem and I thought it would be nice to have it in the message)

Reply:
Removed the extra variable taking the reference, so that now we're sure it will never fail even if we rewrite the code :)